_sources/Iteration/Listsandforloops.rst

```diff
@@ -0,0 +1,154 @@
+
+Remember the turtle drawings we made earlier? For loops and lists can be applied there as well!
+
+Iteration Simplifies our Turtle Program
+---------------------------------------
+
+To draw a square we'd like to do the same thing four times --- move the turtle forward some 
+distance and turn 90 degrees.  We previously used 8 lines of Python code to have alex draw the four 
+sides of a square.  This next program does exactly the same thing but, with the help of the for 
+statement, uses just three lines (not including the setup code).  Remember that the for statement 
+will repeat the `forward` and `left` four times, one time for each value in the list.
+
+.. activecode:: ac6_5_2
+   :nocodelens:
+
+   import turtle            # set up alex
+   wn = turtle.Screen()
+   alex = turtle.Turtle()
+
+   for i in [0, 1, 2, 3]:      # repeat four times
+       alex.forward(50)
+       alex.left(90)
+
+   wn.exitonclick()
+
+
+
+While "saving some lines of code" might be convenient, it is not the big
+deal here.  What is much more important is that we've found a "repeating
+pattern" of statements, and we reorganized our program to repeat the pattern.
+
+The values [0,1,2,3] were provided to make the loop body execute 4 times.
+We could have used any four values.  For example, consider the following program.
+
+
+.. activecode:: ac6_5_3
+   :nocodelens:
+
+   import turtle            # set up alex
+   wn = turtle.Screen()
+   alex = turtle.Turtle()
+
+   for aColor in ["yellow", "red", "purple", "blue"]:      # repeat four times
+       alex.forward(50)
+       alex.left(90)
+
+   wn.exitonclick()
+
+In the previous example, there were four integers in the list.  This time there are four strings. 
+Since there are four items in the list, the iteration will still occur four times.  ``aColor`` will 
+take on each color in the list.  We can even take this one step further and use the value of 
+``aColor`` as part of the computation.
+
+.. activecode:: ac6_5_4
+    :nocodelens:
+
+    import turtle            # set up alex
+    wn = turtle.Screen()
+    alex = turtle.Turtle()
+
+    for aColor in ["yellow", "red", "purple", "blue"]:
+        alex.color(aColor)
+        alex.forward(50)
+        alex.left(90)
+
+    wn.exitonclick()
+
+In this case, the value of ``aColor`` is used to change the color attribute of ``alex``. Each 
+iteration causes ``aColor`` to change to the next value in the list.
+
+The for-loop is our first example of a **compound statement**. Syntactically a compound statement 
+is a statement. The level of indentation of a (whole) compound statement is the indentation of its 
+heading. In the example above there are five statements with the same indentation, executed 
+sequentially: the import, 2 assignments, the *whole* for-loop, and ``wn.exitonclick()``. The 
+for-loop compound statement is executed completely before going on to the next sequential 
+statement, ``wn.exitonclick()``.
+
+**Check your Understanding**
```

Comment:
There are some parsons problems involving the turtle in the thinkcspy book in this section.  Should we add them here?